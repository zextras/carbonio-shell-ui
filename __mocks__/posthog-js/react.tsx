/*
 * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

import React from 'react';

import type * as PostHogReact from 'posthog-js/react';

export const usePostHog: (typeof PostHogReact)['usePostHog'] = () =>
	({
		opt_in_capturing: (): void => undefined,
		opt_out_capturing: (): void => undefined,
		reset: (): void => undefined
	}) as unknown as ReturnType<(typeof PostHogReact)['usePostHog']>;

export const PostHogProvider = ({
	children
}: React.ComponentPropsWithoutRef<(typeof PostHogReact)['PostHogProvider']>): React.JSX.Element => (
	<>{children}</>
);
