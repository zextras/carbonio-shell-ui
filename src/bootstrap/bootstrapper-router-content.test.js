/*
 * *** BEGIN LICENSE BLOCK *****
 * Copyright (C) 2011-2020 Zextras
 *
 *  The contents of this file are subject to the Zextras EULA;
 * you may not use this file except in compliance with the EULA.
 * You may obtain a copy of the EULA at
 * http://www.zextras.com/zextras-eula.html
 * *** END LICENSE BLOCK *****
 */
jest.mock('./bootstrapper-lazy-loader');

import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import BootstrapperRouterContent from './bootstrapper-router-content';

describe('Boostrapper Router Content', () => {
	test('Route to Shell with no Account', () => {
		const { container } = render(
			<MemoryRouter>
				<BootstrapperRouterContent accounts={[]} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});

	test('Route to Shell with Account', () => {
		const { container } = render(
			<MemoryRouter>
				<BootstrapperRouterContent accounts={[{}]} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});

	test('Route to Logout with Account', () => {
		const { container } = render(
			<MemoryRouter initialEntries={['/logout']}>
				<BootstrapperRouterContent accounts={[{}]} />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
