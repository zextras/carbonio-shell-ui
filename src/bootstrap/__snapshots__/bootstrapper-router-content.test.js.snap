// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boostrapper Router Content Route to Logout with Account 1`] = `
<div>
  LazyLogoutView
</div>
`;

exports[`Boostrapper Router Content Route to Shell with Account 1`] = `
<div>
  LazyShellView
</div>
`;

exports[`Boostrapper Router Content Route to Shell with no Account 1`] = `
<div>
  LazyLoginView
</div>
`;
